ui/hooks/useScrollRequired.js

```diff
@@ -12,14 +13,23 @@ import { debounce } from 'lodash';
  */
 export const useScrollRequired = (dependencies = []) => {
   const ref = useRef(null);
+  const previousFirstDependency = usePrevious(dependencies[0]);
+
   const [hasScrolledToBottomState, setHasScrolledToBottom] = useState(false);
   const [isScrollableState, setIsScrollable] = useState(false);
   const [isScrolledToBottomState, setIsScrolledToBottom] = useState(false);
 
+  const resetHasScrolledToBottomWhenFirstDependencyChanges = useCallback(() => {
+    if (!isEqual(previousFirstDependency, dependencies[0])) {
+      setHasScrolledToBottom(false);
+    }
+  }, [dependencies, previousFirstDependency]);
+
   const update = () => {
     if (!ref.current) {
       return;
     }
+    resetHasScrolledToBottomWhenFirstDependencyChanges();
```

Comment:
this feels a bit hacky to me. It does what we need though. Open to other suggestions

context: this code fixes https://github.com/MetaMask/MetaMask-planning/issues/2458
> The confirmation button (Confirm/Sign) should be disabled until the user scrolls to the bottom.

cc: @FrederikBolding

Reply:
another option is we could add setHasScrolledToBottom as a returned value